pallets/circuit/src/lib.rs

```diff
@@ -744,7 +745,7 @@ pub mod pallet {
             max_exec_cost: u128,
             max_notifications_cost: u128,
         ) -> DispatchResultWithPostInfo {
-            let sfx_id = side_effect.generate_id::<SystemHashing<T>>();
+            let sfx_id = side_effect.generate_id::<SystemHashing<T>>(xtx_id.as_ref(), 0u32);
```

Comment:
Im setting sfx_index to 0 here because only one SFX is passed. Is that correct?

Reply:
For now yes, this is YAGNI and not used so will need changing probably later.